src/Arcus.Observability.Telemetry.Core/Extensions/ILoggerExtensions.cs

```diff
@@ -291,17 +291,53 @@ public static void LogServiceBusDependency(this ILogger logger, string entityNam
         /// <param name="entityType">Type of the Service Bus entity</param>
         /// <param name="context">Context that provides more insights on the dependency that was measured</param>
         public static void LogServiceBusDependency(this ILogger logger, string entityName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, ServiceBusEntityType entityType = ServiceBusEntityType.Unknown, Dictionary<string, object> context = null)
         {
             Guard.NotNull(logger, nameof(logger));
+            Guard.NotNullOrWhitespace(entityName, nameof(entityName));
+
+            context = context ?? new Dictionary<string, object>();
 
             logger.LogInformation(ServiceBusDependencyFormat, "Azure Service Bus", entityType, entityName, duration, startTime.ToString(CultureInfo.InvariantCulture), isSuccessful, context);
         }
 
+        /// <summary>
+        ///     Logs an Azure Table Storage Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="tableName">Name of the Table Storage resource</param>
+        /// <param name="accountName">Account of the storage resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="measurement">Measuring the latency to call the Table Storage dependency</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogTableStorageDependency(this ILogger logger, string tableName, string accountName, bool isSuccessful, DependencyMeasurement measurement, Dictionary<string, object> context = null)
+        {
+            Guard.NotNull(logger, nameof(logger));
+
+            LogTableStorageDependency(logger, tableName: tableName, accountName: accountName, isSuccessful: isSuccessful, startTime: measurement.StartTime, duration: measurement.Elapsed, context: context);
+        }
+
+        /// <summary>
+        ///     Logs an Azure Table Storage Dependency.
+        /// </summary>
+        /// <param name="logger">Logger to use</param>
+        /// <param name="tableName">Name of the Table Storage resource</param>
+        /// <param name="accountName">Account of the storage resource</param>
+        /// <param name="isSuccessful">Indication whether or not the operation was successful</param>
+        /// <param name="startTime">Point in time when the interaction with the dependency was started</param>
+        /// <param name="duration">Duration of the operation</param>
+        /// <param name="context">Context that provides more insights on the dependency that was measured</param>
+        public static void LogTableStorageDependency(this ILogger logger, string tableName, string accountName, bool isSuccessful, DateTimeOffset startTime, TimeSpan duration, Dictionary<string, object> context = null)
+        {
```

Comment:
We should guard for empty table and account

Reply:
Yes, done. Thanks.